lab_share_lib/processing/rabbit_message_processor.py

```diff
@@ -60,8 +59,16 @@ def _build_avro_encoders(self, encoder_type: str, subject: str) -> List[AvroEnco
 
     def process_message(self, headers, body):
         message = RabbitMessage(headers, body)
+        subject = message.subject
         try:
-            used_encoder = message.decode(self._build_avro_encoders(message.encoder_type, message.subject))
+            reader_schema_version = self._rabbit_config.message_subjects[subject].reader_schema_version
```

Comment:
Could you move this outside the `try` block, please?  It doesn't do anything that might cause a `TransientRabbitError`

Reply:
There's a test case written to verify the subject is there in `message_subjects`. [It uses](https://github.com/sanger/lab-share-lib/blob/3ee8cb182588306ae76f67384c63ea3f8054933d/tests/processing/test_rabbit_message_processor.py#L185-L191) the `try-catch` block I think.

Reply:
It can't be using the try catch block because the subject is extracted outside of the block

Reply:
Although I suppose what might be happening is that the subject got identified as invalid during the decode process and now it might not exist in the config values.  I think in that case, we would need to catch the `KeyError` when trying to find the subject in the config and do something sensible with that.

Reply:
![image](https://github.com/user-attachments/assets/782d1bc1-228d-4cc3-84c0-a345e21c1c43)

I think it just catches the `KeyError` in the general `Exception` handle and returns false with the correct logging (which is captured by caplog).

Reply:
Ohhh I see.  That catch wasn't visible in the pull request without expanding the code.  I could only see the `TransientRabbitError` block.  I think it would be good to catch the `KeyError` separately in its own try/catch and produce a more useful log message.

Reply:
After all, the current message says the problem occurred during decoding, which isn't true.

Reply:
Yeah I think catching `KeyError` is better.